src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.Application.targets

```diff
@@ -26,6 +27,27 @@ This file contains targets specific for Android application projects.
     </_AndroidComputeRunArgumentsDependsOn>
   </PropertyGroup>
 
+  <!--
+  ***********************************************************************************************
+  ComputeAvailableDevices
+
+  Target that queries available Android devices and emulators.
+  This target is called by 'dotnet run' to support device selection.
+  Returns @(Devices) items with metadata: Description, Type, Status
+
+  See: https://github.com/dotnet/sdk/blob/2b9fc02a265c735f2132e4e3626e94962e48bdf5/documentation/specs/dotnet-run-for-maui.md
+  ***********************************************************************************************
+  -->
+  <Target Name="ComputeAvailableDevices"
+      DependsOnTargets="_ResolveMonoAndroidSdks"
+      Returns="@(Devices)">
+    <GetAvailableAndroidDevices
+        ToolExe="$(AdbToolExe)"
+        ToolPath="$(AdbToolPath)">
+      <Output TaskParameter="Devices" ItemName="Devices" />
+    </GetAvailableAndroidDevices>
+  </Target>
```

Comment:
Is there any possible world where there's some way to do input/output tracking here for incrementality purposes?